PackagingCon2021-fpm/main.tex

```diff
@@ -0,0 +1,320 @@
+      \item single-file version used for creating initial fpm bootstrap binary
+      \item minimal dependencies, requires only a Fortran compiler to build
+      \item bootstrap binary is used to build fpm: Fortran, C compiler and git are required
+   \end{itemize}
+   \vfill
+
+   \centering
+   \includegraphics[width=.9\textwidth]{Figures/bootstrapping.pdf}
+\end{frame}
+
+
+\section{Feature overview}
+
+\begin{frame}{Command-line features}
+   \begin{columns}
+      \hskip.1\textwidth
+      \begin{column}{.6\textwidth}
+         \begin{itemize}
+            \item handles (sub)module interdependencies
+            \item automatically finds executables and tests
+         \end{itemize}
+      \end{column}
+   \end{columns}
+   \vskip2ex
+
+   \begin{columns}
+      \begin{column}{.4\textwidth}
+         \begin{itemize}
+            \item runs executables and examples
+            \item wildcard globbing for selection
+         \end{itemize}
+         \vskip5ex
+
+         \begin{itemize}
+            \item finds and runs test executables
+            \item allows running tests in debugger
+         \end{itemize}
+      \end{column}
+      \begin{column}{.3\textwidth}
+         \includegraphics[width=\textwidth]{Figures/features.pdf}
+      \end{column}
+      \begin{column}{.4\textwidth}
+         \begin{itemize}
+            \item automatically installs project
+            \item user prefix used by default
+         \end{itemize}
+         \vskip5ex
+
+         \begin{itemize}
+            \item updates all dependencies
+            \item basic caching and locking
+         \end{itemize}
+      \end{column}
+   \end{columns}
+   \vskip1ex
+
+   \begin{columns}
+      \hskip.1\textwidth
+      \begin{column}{.6\textwidth}
+         \begin{itemize}
+            \item creates new projects with fpm layout
+            \item backfilling to update existing projects
+         \end{itemize}
+      \end{column}
+   \end{columns}
+\end{frame}
+
+\begin{frame}{Layout of an fpm project}
+   \begin{columns}
+      \begin{column}{.6\textwidth}
+         \includegraphics[width=\textwidth]{Figures/layout.pdf}
+      \end{column}
+      \hskip-1em
+      \begin{column}{.45\textwidth}
+         \begin{itemize}
+            \item package manifest in \alert{TOML}
+            \item limited complexity of build file
+            \item mainly meta data of project\\
+               (license, author, keywords, \ldots)
+               \\[2ex]
+            \item default layout requires only \alert{name}\\
+               (\textcolor{blue}{src}, \textcolor{blue}{app}, \textcolor{blue}{example}, and \textcolor{blue}{test})
+            \item \alert{source-dir} for customizing layout
+               \\[2ex]
+            \item executable names from \alert{program} unit
+               \\[2ex]
+            \item system libraries can be linked
+               \\[2ex]
+         \end{itemize}
+      \end{column}
+   \end{columns}
+\end{frame}
```

Comment:
I recommend acknowledging the inspiration from Rust's Cargo in this slide.

Reply:
I'm probably talking about inspiration from Cargo and Stack at the Implementation slide already.